ember-client/shared/src/main/scala/org/http4s/ember/client/internal/EmberWSClient.scala

```diff
@@ -30,21 +31,35 @@ import org.http4s.client.websocket.WSClient
 import org.http4s.client.websocket.WSConnection
 import org.http4s.client.websocket.WSFrame
 import org.http4s.ember.client.internal.WebSocketHelpers._
-import org.http4s.ember.core.WebSocketHelpers.decodeFrames
-import org.http4s.ember.core.WebSocketHelpers.frameToBytes
+import org.http4s.ember.core.WebSocketHelpers._
+import org.http4s.headers.`Sec-WebSocket-Key`
 import org.http4s.websocket.WebSocketFrame
+import scodec.bits.ByteVector
 
-object EmberWSClient {
+import java.util.Base64
+
+private[client] object EmberWSClient {
   def apply[F[_]](
       emberClient: Client[F]
   )(implicit F: Async[F]): WSClient[F] =
     WSClient[F](respondToPings = false) { wsRequest =>
-      val httpWSRequest = Request[F]()
-        .withUri(wsRequest.uri)
-        .withHeaders(wsRequest.headers)
-        .withMethod(Method.GET)
-
       for {
+        randomByteArray <- Resource.eval(
+          SecureRandom.javaSecuritySecureRandom[F].flatMap(_.nextBytes(16))
+        )
```

Comment:
`SecureRandom.javaSecuritySecureRandom[F]` this will create a new `SecureRandom` every time. Instead we should construct it once, and then re-use it for each request.

```scala
  def apply[F[_]](
      emberClient: Client[F]
  )(implicit F: Async[F]): F[WSClient[F]] =
  SecureRandom.javaSecuritySecureRandom[F].map { random =>
    WSClient[F](respondToPings = false) { wsRequest =>
      // ...
    }
  }
```